src/lazy.ts

```diff
@@ -1,6 +1,28 @@
+const LAZY_SYMBOL = Symbol.for('cdk8s.lazy');
+
 export class Lazy {
   public static any(producer: IAnyProducer): any {
-    return new Lazy(producer) as unknown as any;
+    const lazy = new Lazy(producer) as unknown as any;
+    Object.defineProperty(lazy, LAZY_SYMBOL, { value: true });
```

Comment:
Why not put this in the constructor, like we do for `ApiObject`?

https://github.com/cdk8s-team/cdk8s-core/blob/84a424cdd92773fad764db22b91b6af6bfac8eeb/src/api-object.ts#L166